core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3049,16 +3049,14 @@ public PhysicalOperation visitUpdate(UpdateNode node, LocalExecutionPlanContext
         {
             Integer[] columnValueAndRowIdChannels = new Integer[columnValueAndRowIdSymbols.size()];
             int symbolCounter = 0;
-            // This depends on the outputSymbols being ordered as the blocks of the
-            // resulting page are ordered.
-            for (Symbol symbol : outputSymbols) {
-                int index = columnValueAndRowIdSymbols.indexOf(symbol);
+            for (Symbol symbol : columnValueAndRowIdSymbols) {
+                int index = outputSymbols.indexOf(symbol);
                 if (index >= 0) {
-                    columnValueAndRowIdChannels[index] = symbolCounter;
+                    columnValueAndRowIdChannels[symbolCounter] = index;
                 }
                 symbolCounter++;
             }
-            checkArgument(symbolCounter == columnValueAndRowIdSymbols.size(), "symbolCounter %s should be columnValueAndRowIdChannels.size() %s", symbolCounter);
+            checkArgument(symbolCounter == columnValueAndRowIdSymbols.size(), "symbolCounter %s should be columnValueAndRowIdSymbols.size() %s", symbolCounter, columnValueAndRowIdSymbols.size());
```

Comment:
This check is a noop now. We are iterating over `columnValueAndRowIdSymbols` incrementing `symbolCounter` in every iteration. It will always be true no matter what

Reply:
Agreed; removed.